fs/config/rc.go

```diff
@@ -70,9 +73,18 @@ See the [listremotes](/commands/rclone_listremotes/) command for more informatio
 	})
 }
 
+var rcMatchEnv = regexp.MustCompile(`^RCLONE_CONFIG_(.*?)_TYPE=.*$`)
+
 // Return the a list of remotes in the config file
+// including any defined by environment variables.
 func rcListRemotes(ctx context.Context, in rc.Params) (out rc.Params, err error) {
 	remotes := LoadedData().GetSectionList()
```

Comment:
I was looking to at your other PR #7058 and I realised we already have a function to do this which is called by `rclone listremotes`

https://github.com/rclone/rclone/blob/998d1d1727e5a76b0af2b61bc29a2bd8fa882b9a/cmd/listremotes/listremotes.go#L38

https://github.com/rclone/rclone/blob/998d1d1727e5a76b0af2b61bc29a2bd8fa882b9a/fs/config/config.go#L642-L644

So what I think we should do is call that function here and lose the looking in the environment because that function does it already.

Hopefully your tests won't need any changes

Reply:
It not only fixes env problem in rc but simplifies code indeed. I have made changes following your suggestion.

I do not think that tests need any changes.

Reply:
Thank you